### PR TITLE
[EBPF] Local KMT improvements

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Minor improvements for local KMT usage:

* Enable color output when running commands inside the compiler docker container
* Properly deal with retries in the results of `kmt.test`
* Notify users when pulumi is called and how to show the verbose progress output

### CODE REVIEW

File: tasks/kmt.py [hunk 1]
```diff
@@ -1763,42 +1763,66 @@ def show_last_test_results(ctx: Context, stack: str | None = None):
...
+                if len(testresults) == 1:
+                    result = next(iter(testresults))
+                    if result == "failed":
+                        failures += 1
+                        sum_failures += 1
+                    elif result == "success":
+                        successes += 1
+                    elif result == "skipped":
+                        skipped += 1
+                elif "failed" in testresults and "success" in testresults:
```
Comment (brycekahle) [inline tasks/kmt.py:1801]: What about all retries are failures?

Comment (brycekahle) [reply tasks/kmt.py:1801]: Ah, nevermind. It is a set.

**Review by brycekahle — APPROVED** [review]

File: tasks/kernel_matrix_testing/compiler.py [hunk 2]
```diff
@@ -89,8 +89,12 @@ def exec(self, cmd: str, user="compiler", verbose=True, run_dir: PathOrStr | Non
             cmd = f"cd {run_dir} && {cmd}"
 
         self.ensure_running()
-        return self.ctx.run(
-            f"docker exec -u {user} -i {self.name} bash -c \"{cmd}\"", hide=(not verbose), warn=allow_fail
+
+        # Set FORCE_COLOR=1 so that termcolor works in the container
```
Comment (brycekahle) [inline tasks/kernel_matrix_testing/compiler.py:93]: Kind of related to this. I noticed that if I ran `inv kmt...` without `-e`, I still saw the commands from things running in the container. Is there a way we can detect the `echo` setting and propagate it correctly?

Comment (gjulianm) [reply tasks/kernel_matrix_testing/compiler.py:93]: Fixed!

**Pull request merged**